### PR TITLE
Feature/ Add byte[] download methods

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# Description

Add two methods for handling byte[] downloading:

* Added `DownloadFileBytesAsync` method - Downloads a file (with caching) and returns the byte[] of the file.
* Added `DownloadBytesAsync` method - Only downloads and returns the byte[] from a url, no caching.

### CODE REVIEW

File: Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs [hunk 1]
```diff
@@ -983,6 +983,67 @@ public static bool TryGetFileNameFromUrl(string url, out string fileName)
...
+        string fileName = null,
+        RestParameters parameters = null,
+        bool debug = false,
+        CancellationToken cancellationToken = default)
+        {
+            await Awaiters.UnityMainThread;
+            byte[] bytes = null;
+
+            var filePath = await DownloadFileAsync(url, fileName, parameters, false, cancellationToken);
+            var absolutefilePath = filePath.Replace("file://", "");
```
Comment (StephenHodgson) [inline Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs]:
```suggestion
            var absolutefilePath = filePath.Replace("file://", string.Empty);
```

File: Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs [hunk 2]
```diff
@@ -983,6 +983,67 @@ public static bool TryGetFileNameFromUrl(string url, out string fileName)
...
+            if (File.Exists(absolutefilePath))
```
Comment (StephenHodgson) [inline Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs]:
```suggestion

            if (File.Exists(absolutefilePath))
```

File: Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs [hunk 3]
```diff
@@ -983,6 +983,67 @@ public static bool TryGetFileNameFromUrl(string url, out string fileName)
...
+            {
+                try
+                {
+                    bytes = File.ReadAllBytes(absolutefilePath);
+                }
+                catch (Exception ex)
+                {
+                    Debug.LogError(ex);
+                    throw;
```
Comment (StephenHodgson) [inline Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs]: If you're going to throw I think it's okay to remove log and try catch

File: Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs [hunk 4]
```diff
@@ -983,6 +983,67 @@ public static bool TryGetFileNameFromUrl(string url, out string fileName)
...
+        string url,
+        string fileName = null,
+        RestParameters parameters = null,
+        bool debug = false,
+        CancellationToken cancellationToken = default)
+        {
+            await Awaiters.UnityMainThread;
+            byte[] bytes = null;
+
+            var filePath = await DownloadFileAsync(url, fileName, parameters, false, cancellationToken);
```
Comment (StephenHodgson) [inline Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs]: This prob won't work on WebGL, and we should fall back to `DownloadBytesAsync`

**Review by StephenHodgson — CHANGES_REQUESTED** [review]
See comments

File: Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs (hunk 3 above)
Comment (SimonDarksideJ) [reply Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs]: Simply modeled on how try was handled in other parts of the code, if you have a suggestion?

File: Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs (hunk 4 above)
Comment (SimonDarksideJ) [reply Utilities.Rest/Packages/com.utilities.rest/Runtime/Rest.cs]: If you are on WebGL, then you would not use it, so why have a fallback?
WebGL would only use DownloadBytes

**Review by StephenHodgson — APPROVED** [review]

**Pull request merged**